model/utils.go

```diff
@@ -127,6 +137,15 @@ func (m *StringMap) Scan(value any) error {
 func (m StringMap) Value() (driver.Value, error) {
 	ok := m[BinaryParamKey]
 	delete(m, BinaryParamKey)
+
+	sz := 0
+	for k := range m {
+		sz += len(k) + len(m[k])
```

Comment:
> You cannot measure the size of a map this way

Can you explain this a little bit. I mean both keys and values are strings in this map.

Reply:
I mean, you are missing the `"`, and `:` etc. It's just simpler and more readable to calculate the length of the marshaled slice.

Reply:
I initially tried with the slice, but the memory jumps somewhere twice relative to this approach.
Although it will be GC'd, but I thought if we could avoid that, we should.
However, if you feel it won't matter that much, I can change it to the marshal-way.

Reply:
I see. Okay, let's use marshal, but use a limit writer to limit the number of bytes marshalled. This avoids any redundant CPU computation taken due to looping through the entire map, and also limits the bytes marshalled at the same time. We just need to use a `json.NewEncoder(limitedWriter).Encode(map)` and return the error from the writer.

Reply:
Even with a limitedWriter, it is still _almost_ the same result. As it still needs to marshal it before using the "limitedWriter". But again, it all goes away after GC.
So, to keep things simpler, I think we can go with the marshaled-slice approach?

Reply:
> As it still needs to marshal it before using the "limitedWriter".

Not really, you will be marshalling to the writer, which won't allow bytes to be written beyond a limit.

Can you show me the code?

Reply:
> We just need to use a json.NewEncoder(limitedWriter).Encode(map)

Here `Encode` runs first which does the marshalling before calling the Writer. 
marshal - https://cs.opensource.google/go/go/+/refs/tags/go1.18.4:src/encoding/json/stream.go;l=206
write - https://cs.opensource.google/go/go/+/refs/tags/go1.18.4:src/encoding/json/stream.go;l=231

The code I tried
```go
type LimitedWriter struct {
	buf  *bytes.Buffer
	lmt int
}

func (w *LimitedWriter) Write(p []byte) (n int, err error) {
	if len(p)+w.buf.Len() > w.lmt {
		return 0, errors.New("Overflow")
	}
	return w.buf.Write(p)
}

w := &LimitedWriter{buf: &bytes.Buffer{}, lmt: maxPropSizeBytes}
err := json.NewEncoder(w).Encode(m)
```

Reply:
You are right. I missed the fact that `Encode` marshals the whole object first and then writes. Sorry for the churn. Let's use your original approach then. Unfortunately we can't do it for `StringInterface`, so it's okay to do it post marshalling there.

Let's just change the error to what I suggested here: https://github.com/mattermost/mattermost-server/pull/20900#discussion_r966600580.

Thanks!